docs/html/conf.py

```diff
@@ -12,22 +12,30 @@
 sys.path.insert(0, docs_dir)
 
 # -- General configuration ------------------------------------------------------------
-
 extensions = [
-    # first-party extensions
-    "sphinx.ext.autodoc",
-    "sphinx.ext.todo",
-    "sphinx.ext.extlinks",
-    "sphinx.ext.intersphinx",
-    # our extensions
+    # extensions common to 'man' and 'html' builds
     "pip_sphinxext",
-    # third-party extensions
-    "myst_parser",
-    "sphinx_copybutton",
-    "sphinx_inline_tabs",
-    "sphinxcontrib.towncrier",
 ]
 
+# 'tags' is a special object handled by sphinx
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-tags
+if "html" in tags or "READTHEDOCS" in os.environ:  # type: ignore[name-defined] # noqa: F821
+    # html specific deps
+    extensions.extend(
+        (
+            # first-party extensions
+            "sphinx.ext.autodoc",
+            "sphinx.ext.todo",
+            "sphinx.ext.extlinks",
+            "sphinx.ext.intersphinx",
+            # third-party extensions
+            "myst_parser",
+            "sphinx_copybutton",
+            "sphinx_inline_tabs",
```

Comment:
These two seem to be the only genuine 'HTML-only' extensions. Perhaps the condition might be better expressed as `'man' not in tags`?

You could also look at the Python docs, where we use a pattern to check if an extension is importable, and if so add it to `extensions`. Eg 

```
import importlib.util
if importlib.util.find_spec(...):
    extensions.append(...)
```

Reply:
> Perhaps the condition might be better expressed as 'man' not in tags?

:+1: this was also suggested above (https://github.com/pypa/pip/pull/13168#issuecomment-2594029703) and I agree it makes this clearer (and simplifies the code!) so trying that: 033c31c9ff4d95695594bfaf2d59d7f34ce5134b